client/keys/list.go

```diff
@@ -39,6 +39,10 @@ func runListCmd(cmd *cobra.Command, _ []string) error {
 	}
 
 	for _, r := range records {
+		if r == nil {
```

Comment:
Since I set capacity on `Record` slice [here](https://github.com/cosmos/cosmos-sdk/blob/cyberbono3/keyring-migration/crypto/keyring/keyring.go#L498), some items in `records` are nil. Hence, I have to print out the name of non empty `Record` at L46.

Reply:
Please don't do that. It's error prone that we return `nil` keys in `Keyring.List()`

Reply:
Sure! `Keyring.List()` does not return nil items.